src/apps/mixed_fem_helper.hpp

```diff
@@ -0,0 +1,942 @@
+    qpts1D[1] = 0.788675134594812882;
+
+    // Order 2 Gauss-Legendre Weights
+    wgts1D[0] = 0.5;
+    wgts1D[1] = 0.5;
+
+    break;
+
+  case 3 :
+
+    // Order 3 Gauss-Legendre Points
+    qpts1D[0] = 0.112701665379258311;
+    qpts1D[1] = 0.5;
+    qpts1D[2] = 0.887298334620741689;
+
+    // Order 3 Gauss-Legendre Weights
+    wgts1D[0] = 0.277777777777777778;
+    wgts1D[1] = 0.444444444444444444;
+    wgts1D[2] = 0.277777777777777778;
+
+    break;
+
+  case 4 :
+
+    // Order 4 Gauss-Legendre Points
+    qpts1D[0] = 0.0694318442029737124;
+    qpts1D[1] = 0.330009478207571868;
+    qpts1D[2] = 0.669990521792428132;
+    qpts1D[3] = 0.930568155797026288;
+
+    // Order 4 Gauss-Legendre Weights
+    wgts1D[0] = 0.173927422568726929;
+    wgts1D[1] = 0.326072577431273071;
+    wgts1D[2] = 0.326072577431273071;
+    wgts1D[3] = 0.173927422568726929;
+
+    break;
+
+  case 5 :
+
+    // Order 5 Gauss-Legendre Points
+    qpts1D[0] = 0.0469100770306680036;
+    qpts1D[1] = 0.230765344947158454;
+    qpts1D[2] = 0.5;
+    qpts1D[3] = 0.769234655052841546;
+    qpts1D[4] = 0.953089922969331996;
+
+    // Order 5 Gauss-Legendre Weights
+    wgts1D[0] = 0.118463442528094544;
+    wgts1D[1] = 0.239314335249683234;
+    wgts1D[2] = 0.284444444444444444;
+    wgts1D[3] = 0.239314335249683234;
+    wgts1D[4] = 0.118463442528094544;
+
+    break;
+
+  }
+
+}
+
+RAJA_HOST_DEVICE
+inline void get_quadrature_rule(
+  const int    quad_type,
+  const int    quad_order,
+  double       (&qpts_1d)[MAX_QUAD_ORDER],
+  double       (&wgts_1d)[MAX_QUAD_ORDER])
+{
+  // Generate the 1D set of points and weights over the interval [0,1]
+  switch( quad_type ) {
+
+  case 0 :
+    LinAlg_qrule_Lobatto(quad_order, qpts_1d, wgts_1d);
+    break;
+
+  case 1 :
+    LinAlg_qrule_Legendre(quad_order, qpts_1d, wgts_1d);
+    break;
+
+  }
+}
+
+RAJA_HOST_DEVICE
+constexpr double compute_detj(
+  const double jxx,
+  const double jxy,
+  const double jxz,
+  const double jyx,
+  const double jyy,
+  const double jyz,
+  const double jzx,
+  const double jzy,
+  const double jzz)
+{
+  return
+    jxy*jyz*jzx - jxz*jyy*jzx + jxz*jyx*jzy -
+    jxx*jyz*jzy - jxy*jyx*jzz + jxx*jyy*jzz;
+}
+
+template<int M>
+RAJA_HOST_DEVICE
+constexpr void transform_basis(
+  const double txx,
+  const double txy,
+  const double txz,
+  const double tyx,
+  const double tyy,
+  const double tyz,
+  const double tzx,
+  const double tzy,
+  const double tzz,
+  const double (&basis_x)[M],
+  const double (&basis_y)[M],
+  const double (&basis_z)[M],
+  double (&tbasis_x)[M],
+  double (&tbasis_y)[M],
+  double (&tbasis_z)[M])
+{
+  // Compute transformed basis function gradients
+  for (int m = 0; m < M; m++)
+  {
+    tbasis_x[m] = txx*basis_x[m] + txy*basis_y[m] + txz*basis_z[m];
+    tbasis_y[m] = tyx*basis_x[m] + tyy*basis_y[m] + tyz*basis_z[m];
+    tbasis_z[m] = tzx*basis_x[m] + tzy*basis_y[m] + tzz*basis_z[m];
+  }
+}
+
+template<int M, int P>
```

Comment:
Would this make more sense as constexpr instead of a non type template parameter?

Reply:
Resolved in person